org.eclipse.search/search/org/eclipse/search/internal/ui/SearchPreferencePage.java

```diff
@@ -56,12 +56,15 @@ public class SearchPreferencePage extends FieldEditorPreferencePage implements I
 	public static final String TEXT_SEARCH_ENGINE = "org.eclipse.search.textSearchEngine"; //$NON-NLS-1$
 	public static final String TEXT_SEARCH_QUERY_PROVIDER = "org.eclipse.search.textSearchQueryProvider"; //$NON-NLS-1$
 	public static final String LIMIT_HISTORY= "org.eclipse.search.limitHistory"; //$NON-NLS-1$
+	public static final String MAX_RESULTS = "org.eclipse.search.maxResults"; //$NON-NLS-1$
+
+	private static final int MAX_RESULTS_DEFAULT = 2000;
```

Comment:
I would prefer a far higher default. Like the count when OOME or lags start. At which number do you get trouble?

Reply:
Fully agree here, 2000 is too small. 100.000 **at least** would be OK I guess.

Reply:
The amount depends on the users workstation. As mentioned on my original report, my system had completely failed by 500k, it had been completely unresponsive way before then like at about 250k. I had only gone up to 500k because I was trying to stop / close eclipse normally. In the end a kill -9 was required for me. For reference my cpu was an i5-8250u, 2gb of ram for eclipse jvm, and it was all on an ssd.

Note that the count / memory is only part of the problem. The code base I was working on was very small. It was only about 1000 files, 50k lines of code, total source size of about 2mb of text files. The problem was most likely the amplification cause by the wildcard search matching literally everything in every file many times. 

I think any default size beyond 100k would defeat the purpose of this feature as by default, it would no longer provide the circuit breaker functionality of stopping a run away query. In many cases, I can no longer stop a query even after it hits just the 10k's, the best I can do often is wait for it to finish.

So given this discussion, I'm no longer confident that what I've submitted is of any use to either you guys nor myself at this point. I'm not sure where to go from here.